src/nvim/tui/tui.c

```diff
@@ -576,11 +576,30 @@ static void tui_set_mode(UI *ui, ModeShape mode)
                  data->params, out, ui, NULL, NULL);
   }
 
-  if (c.id != 0 && ui->rgb) {
+  // update cursor colors
+  if (c.id != 0) {
     int attr = syn_id2attr(c.id);
     attrentry_T *aep = syn_cterm_attr2entry(attr);
-    data->params[0].i = aep->rgb_bg_color;
-    unibi_out(ui, data->unibi_ext.set_cursor_color);
+
+    if (ui->rgb) {
+
+      RgbValue bg = aep->rgb_bg_color;
```

Comment:
Is this more succinct using a ternary?

```c
// if gui=reverse, swap bg and fg colors
RgbValue bg = aep->rgb_ae_attr == HL_INVERSE ? aep->rgb_fg_color : aep->rgb_bg_color;
```
Just a random thought..might not be worth changing.